api/services/introspection/introspection.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+package containerd.v1;
+
+import "gogoproto/gogo.proto";
+
+service Introspection {
+	rpc Introspect(IntrospectRequest) returns (IntrospectResponse);
+}
+
+message Version {
+	string version = 1;
+	string revision = 2;
+}
+
+enum PluginType {
+	UNKNOWN = 0;
+	RUNTIME = 1;
+	GRPC = 2;
+	SNAPSHOT = 3;
+	CONTAINER_MONITOR = 4;
+}
+
+message PluginRegistration {
+	string name = 1;
+	PluginType type = 2;
+}
+
+enum IntrospectionSelectorType {
```

Comment:
Let's just make separate methods for this, rather than implementing an adhoc mechanism.